flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/SchemaCoder.java

```diff
@@ -34,6 +35,20 @@ public interface SchemaCoder {
 
     void writeSchema(Schema schema, OutputStream out) throws IOException;
 
+    default void writeSchema(
+            Schema schema,
+            OutputStream out,
+            Map<String, Object> inputProperties,
+            Map<String, Object> outputProperties)
+            throws IOException {
+        throw new RuntimeException("writeSchema passing headers should be overridden.");
```

Comment:
why is this not calling `writeSchema(schema, out)` like you do with `readSchemaWithAdditionalParameters`?